### PR TITLE
Move stats to impl.stats, change some visibility.

### PR DESCRIPTION
**Pull request by songy23**
Will remove `StatsContextFactory`, `Tag`, `TagKey`, `TagValue` under stats package, and use new APIs in tags package in the next PR.

### CODE REVIEW

File: core/src/main/java/io/opencensus/stats/AggregationData.java [hunk 1]
```diff
@@ -130,10 +130,16 @@ static CountData create(long count) {
     HistogramData() {
     }
 
-    static HistogramData create(long... bucketCountDatas) {
-      checkNotNull(bucketCountDatas, "bucket counts should not be null.");
+    /**
+     * Create a {@code HistogramData}.
```
Comment (sebright) [inline core/src/main/java/io/opencensus/stats/AggregationData.java]: The summary fragment should be an incomplete sentence, such as "`Creates a {@code HistogramData}.`" (https://google.github.io/styleguide/javaguide.html#s7.2-summary-fragment).  This applies to RangeData, too.

File: core/src/main/java/io/opencensus/stats/AggregationData.java [hunk 2]
```diff
@@ -66,7 +66,7 @@ private AggregationData() {
     SumData() {
     }
 
-    static SumData create(double sum) {
+    public static SumData create(double sum) {
```
Comment (sebright) [inline core/src/main/java/io/opencensus/stats/AggregationData.java]: All of the public factory methods in this class should have Javadocs.

Comment (songy23) [inline core/src/main/java/io/opencensus/stats/AggregationData.java]: Added.

File: core/src/main/java/io/opencensus/stats/AggregationData.java (hunk 1 above)
Comment (songy23) [inline core/src/main/java/io/opencensus/stats/AggregationData.java]: Updated. Thanks for pointing out.

**Comment by songy23** [comment]
@bogdandrutu `tags` impl has already been moved to `impl/tags`: https://github.com/census-instrumentation/opencensus-java/tree/master/core_impl/src/main/java/io/opencensus. This PR is meant to move `stats` impl package to the same location with `tags`.

**Comment by bogdandrutu** [comment]
It will be incompatible with Trace so @dinooliva can you make a call if that is acceptable?

**Comment by songy23** [comment]
Rebased against mainline.

**Comment by bogdandrutu** [comment]
Trace changed to the same implementation so it is fine to merge this PR.

**Comment by songy23** [comment]
Got it. Rebased again... should be ready to merge now.

File: core/src/main/java/io/opencensus/stats/StatsRecorder.java [hunk 3]
```diff
@@ -33,7 +33,7 @@
    * @param tags the tags associated with the measurements.
    * @param measurementValues the measurements to record.
    */
-  abstract void record(TagContext tags, MeasureMap measurementValues);
+  protected abstract void record(TagContext tags, MeasureMap measurementValues);
```
Comment (dinooliva) [inline core/src/main/java/io/opencensus/stats/StatsRecorder.java]: Should this be public?

Comment (sebright) [inline core/src/main/java/io/opencensus/stats/StatsRecorder.java]: I think this can be public now that we've switched to TagContext.

Comment (songy23) [inline core/src/main/java/io/opencensus/stats/StatsRecorder.java]: Done.

**Comment by songy23** [comment]
Thanks!

**Pull request merged**